src/groups/mqb/mqba/mqba_clientsession.cpp

```diff
@@ -2770,38 +2757,62 @@ void ClientSession::processEvent(
                 return;  // RETURN
             }
 
-            d_beginTimestamp = bmqsys::Time::highResolutionTimer();
+            bsl::shared_ptr<bmqsys::OperationLogger> opLogger =
+                bsl::allocate_shared<bmqsys::OperationLogger>(
+                    d_state.d_allocator_p);
+            opLogger->start() << description() << ": disconnect";
 
             d_isDisconnecting = true;
             eventCallback     = bdlf::BindUtil::bind(
                 &ClientSession::processDisconnectAllQueues,
                 this,
-                controlMessage);
+                controlMessage,
+                opLogger);
         } break;
         case MsgChoice::SELECTION_ID_OPEN_QUEUE: {
-            d_beginTimestamp = bmqsys::Time::highResolutionTimer();
+            const bsl::string& queueUri =
+                controlMessage.choice().openQueue().handleParameters().uri();
+
+            bsl::shared_ptr<bmqsys::OperationLogger> opLogger =
+                bsl::allocate_shared<bmqsys::OperationLogger>(
+                    d_state.d_allocator_p);
+            opLogger->start()
+                << description() << ": open queue [" << queueUri << "]";
 
             eventCallback = bdlf::BindUtil::bind(
                 &ClientSession::processOpenQueue,
                 this,
-                controlMessage);
+                controlMessage,
+                opLogger);
         } break;
         case MsgChoice::SELECTION_ID_CLOSE_QUEUE: {
-            d_beginTimestamp = bmqsys::Time::highResolutionTimer();
+            const bsl::string& queueUri =
+                controlMessage.choice().closeQueue().handleParameters().uri();
+
+            bsl::shared_ptr<bmqsys::OperationLogger> opLogger =
+                bsl::allocate_shared<bmqsys::OperationLogger>(
+                    d_state.d_allocator_p);
+            opLogger->start()
+                << description() << ": close queue [" << queueUri << "]";
 
             eventCallback = bdlf::BindUtil::bind(
                 &ClientSession::processCloseQueue,
                 this,
-                controlMessage);
+                controlMessage,
+                opLogger);
         } break;
         case MsgChoice::SELECTION_ID_CONFIGURE_QUEUE_STREAM:
         case MsgChoice::SELECTION_ID_CONFIGURE_STREAM: {
-            d_beginTimestamp = bmqsys::Time::highResolutionTimer();
+            bsl::shared_ptr<bmqsys::OperationLogger> opLogger =
+                bsl::allocate_shared<bmqsys::OperationLogger>(
+                    d_state.d_allocator_p);
```

Comment:
Note: I wanted to use object pool to reuse operation loggers, however, this is not a good approach here.
1. `MemOutStream` used in `bmqsys::OperationLogger` doesn't reuse the internal buffer, so we don't benefit from reusing the same object. `MemOutStream::reset()` frees the internal buffer completely, and we don't have other way to save it. `bsl::stringstream` doesn't have this either.
2. `mqba::ClientSession` has a non-trivial lifetime. There can be dispatcher event associated with client sessions that were already destructed. If client session owns an object pool, we cannot return items to this object pool after client session destruction.